bridges/AnidexBridge.php

```diff
@@ -0,0 +1,205 @@
+					'Portuguese (Portugal)' => '17',
+					'Swedish' => '18',
+					'Arabic' => '19',
+					'Danish' => '20',
+					'Chinese (Simplified)' => '21',
+					'Bengali' => '22',
+					'Romanian' => '23',
+					'Czech' => '24',
+					'Mongolian' => '25',
+					'Turkish' => '26',
+					'Indonesian' => '27',
+					'Korean' => '28',
+					'Spanish (LATAM)' => '29',
+					'Persian' => '30',
+					'Malaysian' => '31'
+				)
+			),
+			'group_id' => array(
+				'name' => 'Group ID',
+				'type' => 'number'
+			),
+			'r' => array(
+				'name' => 'Hide Remakes',
+				'type' => 'checkbox'
+			),
+			'b' => array(
+				'name' => 'Only Batches',
+				'type' => 'checkbox'
+			),
+			'a' => array(
+				'name' => 'Only Authorized',
+				'type' => 'checkbox'
+			),
+			'q' => array(
+				'name' => 'Keyword',
+				'description' => 'Keyword(s)',
+				'type' => 'text'
+			),
+			'h' => array(
+				'name' => 'Adult content',
+				'type' => 'list',
+				'values' => array(
+					'No filter' => '0',
+					'Hide +18' => '1',
+					'Only +18' => '2'
+				)
+			)
+		)
+	);
+
+	public function collectData() {
+
+		// Build Search URL from user-provided parameters
+		$search_url = self::URI . '?s=upload_timestamp&o=desc';
+		foreach (array('id', 'lang_id', 'group_id') as $param_name) {
+			$param = $this->getInput($param_name);
+			if (!empty($param) && intval($param) != 0 && ctype_digit(str_replace(',', '', $param))) {
+				$search_url .= '&' . $param_name . '=' . $param;
+			}
+		}
+		foreach (array('r', 'b', 'a') as $param_name) {
+			$param = $this->getInput($param_name);
+			if (!empty($param) && boolval($param)) {
+				$search_url .= '&' . $param_name . '=1';
+			}
+		}
+		$query = $this->getInput('q');
+		if (!empty($query)) {
+			$search_url .= '&q=' . urlencode($query);
+		}
+		$opt = array();
+		$h = $this->getInput('h');
+		if (!empty($h) && intval($h) != 0 && ctype_digit($h)) {
+			$opt[CURLOPT_COOKIE] = 'anidex_h_toggle=' . $h;
+		}
+
+		// Retrieve torrent listing from search results, which does not contain torrent description
+		$html = getSimpleHTMLDOM($search_url, array(), $opt)
+		or returnServerError('Could not request Anidex: '.$search_url);
+		$links = $html->find('a');
+		$results = array();
+		foreach ($links as $link)
+			if (strpos($link->href, '/torrent/') === 0 && !in_array($link->href, $results))
+				$results[] = $link->href;
+		if (empty($results))
+			returnServerError('No results from Anidex: '.$search_url);
+		$limit = 0;
+
+		//Process each item individually
+		foreach ($results as $element) {
+
+			$torrent_id = str_replace('/torrent/', '', $element);
+
+			//Limit total amount of requests and ignore entries without valid torrent ID
+			if ($limit < 20 && $torrent_id != 0 && ctype_digit($torrent_id)) {
+
+				//Retrieve data for this torrent ID
+				$item_uri = self::URI . 'torrent/'.$torrent_id;
+
+				//Retrieve full description from torrent page
+				if ($item_html = getSimpleHTMLDOMCached($item_uri)) {
```

Comment:
You should fail if the condition is false, else it might enter an infinite loop (or at least move the $limit++ up)

Reply:
This bridge works by retrieving all links in the page and processing only links containing a specific pattern, in a limit of 20. Some links *will* not pass the conditions and are ignored on purpose. I think this behavior is not likely to cause an infinite loop since the amount of links in the page would need to be infinite too and Simple DOM Parser would fail before that. How about moving `$limit++` right before calling `getSimpleHTMLDOMCached`?

Reply:
See my comment below. You can simply check for the limit after adding items to `$this->items`:

```PHP
if(count($this->items) >= 10) {
    break;
}
```

That way, either the limit is reached and the loop ends, or the loop continues until all items were checked and terminates normally.

Reply:
Thanks, changing to that. :+1: